.github/workflows/github-actions-demo.yml

```diff
@@ -0,0 +1,26 @@
+name: leaderboard-backend CI
```

Comment:
Discussion point for here and below: this name feels very redundant. Do other repos name these specific things?

Reply:
Also file naming. This works and runs code, so it shouldn't be treated as a "demo".

Reply:
Generally I believe actions get descriptive names such as "[build-test](https://github.com/actions/setup-go/blob/main/.github/workflows/workflow.yml)" or "[Licensed](https://github.com/actions/setup-go/blob/main/.github/workflows/licensed.yml)" or "[Validate 'setup-go'](https://github.com/actions/setup-go/blob/main/.github/workflows/versions.yml)" - all of which are examples from the github actions available actions.

Reply:
renamed to `backend-test` and `backend-lint` to be more useful